### PR TITLE
Dem 68 connect backend

### PR DESCRIPTION
**Pull request by indytechcook**
The is the first pass at using an API Resource in a service.  The is just a start and will be refactor but this commit was getting big and I wanted to break it up more.

@tladendo @vietnogi

### CODE REVIEW

File: gulp.config.js [hunk 1]
```diff
@@ -25,7 +25,8 @@ module.exports = function () {
     // angular contants
     ngConstants: {
       apiUrl: envConfig.getVal('BASE_API_URL', '/v3/'),
-      useStubs: useStubs
+      useStubs: useStubs,
+      apiToken: envConfig.getVal('JWT_TOKEN', '')
```
Comment (indytechcook) [inline gulp.config.js]: This is used for testing against the API until we get the auth working.

File: src/client/app/core/resource/resource.data.provider.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
+(function() {
```
Comment (indytechcook) [inline src/client/app/core/resource/resource.data.provider.js]: This provider will be used mostly for router resolving.

File: src/client/app/work-request/work-request.module.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  angular
+    .module('app.workRequest', [
+      'blocks.exception', 'blocks.logger',
+      'app.resource'
+    ])
+    .run(WorkRequest);
+
+  WorkRequest.$inject = ['ApiResource'];
+
+  function WorkRequest(ApiResource) {
```
Comment (indytechcook) [inline src/client/app/work-request/work-request.module.js]: This is how new api resources are added to the API.

**Comment by tladendo** [comment]
lgtm

**Pull request merged**